bundles/org.openhab.core.config.core/src/main/java/org/openhab/core/config/core/OrderingMapSerializer.java

```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.config.core;
+
+import java.lang.reflect.Type;
+import java.util.Comparator;
+import java.util.Map;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonSerializationContext;
+import com.google.gson.JsonSerializer;
+
+/**
+ * Serializes map data by ordering the keys
+ *
+ * @author Sami Salonen - Initial contribution
+ */
+@NonNullByDefault
+public class OrderingMapSerializer implements JsonSerializer<Map<String, Object>> {
```

Comment:
GsonBuilder.registerTypeHierarchyAdapter(Map.class, new OrderingMapSerializer()) can result in OrderingMapSerializer to be used with non-Comparable keys and thus with runtime exception -- even though this class by itself looks OK (String keys are comparable)

Reply:
Resolved in Resolved in https://github.com/openhab/openhab-core/pull/2437/commits/69f4c812c434e138dca67f0e2e0d7dbd21490886